core/pta/system.c

```diff
@@ -104,14 +114,27 @@ static TEE_Result system_derive_ta_unique_key(struct user_mode_ctx *uctx,
 	memcpy(data, &uctx->ts_ctx->uuid, sizeof(TEE_UUID));
 
 	/* Append the user provided data */
-	memcpy(data + sizeof(TEE_UUID), params[0].memref.buffer,
-	       params[0].memref.size);
+	res = copy_from_user(data + sizeof(TEE_UUID), params[0].memref.buffer,
+			     params[0].memref.size);
+	if (res)
+		goto out;
+
+	res = bb_memdup_user(params[1].memref.buffer, params[1].memref.size,
```

Comment:
This is an out buffer so `bb_alloc()` should be used instead.

Reply:
updated